Fix stale pull_artifact example API usage

## Summary

The Rust SDK example `rust/ommx/examples/pull_artifact.rs` no longer matches the current artifact API. It imports `pull_image` from `ommx::artifact::local_registry`, but remote pulls are now exposed as the `LocalRegistry::pull_image` method.

## Reproduction

```bash
cargo check -p ommx --example pull_artifact --features remote-artifact,cli
```

## Current error

```text
error[E0432]: unresolved import `ommx::artifact::local_registry::pull_image`
 --> rust/ommx/examples/pull_artifact.rs:3:22
  |
3 |     local_registry::{pull_image, LocalRegistry},
  |                      ^^^^^^^^^^ no `pull_image` in `artifact::local_registry`
```

## Expected fix

Update the example to use the current method form:

```rust
let registry = LocalRegistry::shared_default()?;
registry.pull_image(&image_name)?;
```

and remove the stale `pull_image` import.

--- a/rust/ommx/examples/pull_artifact.rs
+++ b/rust/ommx/examples/pull_artifact.rs
@@ -1,8 +1,5 @@
 use anyhow::Result;
-use ommx::artifact::{
-    local_registry::{pull_image, LocalRegistry},
-    media_types, ImageRef, LocalArtifact,
-};
+use ommx::artifact::{local_registry::LocalRegistry, media_types, ImageRef, LocalArtifact};
 
 fn main() -> Result<()> {
     tracing_subscriber::fmt()
@@ -18,7 +15,7 @@ fn main() -> Result<()> {
     // Pull the artifact from the remote registry into the v3 SQLite
     // Local Registry, then open it for read by ref.
     let registry = LocalRegistry::shared_default()?;
-    pull_image(registry, &image_name)?;
+    registry.pull_image(&image_name)?;
     let local = LocalArtifact::open_in_registry(registry, image_name)?;
 
     // Print the digest of each instance layer.

